### PR TITLE
chore: Update Field Spec and Migration guide to match current implementation

### LINKED ISSUE #26548: chore: Update Field spec to match current implementation
Need to update Field's spec to match the implementation changes from PR #26430. Field is now a standalone component instead of being integrated into `InputField`, etc. components.

### PR DESCRIPTION
**Pull request by behowell**
## Previous Behavior

The Field spec is outdated because of the implementation changes to make it a standalone component in PR #26430.

## New Behavior

Update the Spec and Migration guide to match the new implementation.

## Related Issue(s)

* Fixes #26548

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/Checkbox.fixture.js'>Checkbox</abbr>                                                                              |  `32.768 kB`<br />`10.213 kB` |
| <samp>react-checkbox</samp> <br /> <abbr title='bundle-size/CheckboxField.fixture.js'>CheckboxField</abbr>                                                                    |  `39.673 kB`<br />`12.094 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |  `82.678 kB`<br />`26.408 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/ComboboxField.fixture.js'>ComboboxField</abbr>                                                                    |  `79.095 kB`<br />`26.183 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `81.346 kB`<br />`26.144 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr>              |  `60.588 kB`<br />`16.969 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> |  `188.355 kB`<br />`52.98 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.751 kB`<br />`11.453 kB` |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr>                                                                                       |   `17.605 kB`<br />`6.518 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/Input.fixture.js'>Input</abbr>                                                                                       |   `24.848 kB`<br />`8.136 kB` |
| <samp>react-input</samp> <br /> <abbr title='bundle-size/InputField.fixture.js'>InputField</abbr>                                                                             |   `34.99 kB`<br />`11.072 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `6.069 kB`<br />`2.053 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressBar.fixture.js'>ProgressBar</abbr>                                                                        |   `13.573 kB`<br />`5.232 kB` |
| <samp>react-progress</samp> <br /> <abbr title='bundle-size/ProgressField.fixture.js'>ProgressField</abbr>                                                                    |   `23.805 kB`<br />`8.339 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/Radio.fixture.js'>Radio</abbr>                                                                                       |  `32.353 kB`<br />`10.504 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroup.fixture.js'>RadioGroup</abbr>                                                                             |    `14.822 kB`<br />`5.91 kB` |
| <samp>react-radio</samp> <br /> <abbr title='bundle-size/RadioGroupField.fixture.js'>RadioGroupField</abbr>                                                                   |   `24.895 kB`<br />`9.045 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/Select.fixture.js'>Select</abbr>                                                                                    |    `24.14 kB`<br />`8.271 kB` |
| <samp>react-select</samp> <br /> <abbr title='bundle-size/SelectField.fixture.js'>SelectField</abbr>                                                                          |  `33.316 kB`<br />`10.755 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr>                                                                                    |  `32.768 kB`<br />`10.438 kB` |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/SliderField.fixture.js'>SliderField</abbr>                                                                          |  `42.599 kB`<br />`13.345 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButton.fixture.js'>SpinButton</abbr>                                                                        |  `45.161 kB`<br />`13.035 kB` |
| <samp>react-spinbutton</samp> <br /> <abbr title='bundle-size/SpinButtonField.fixture.js'>SpinButtonField</abbr>                                                              |  `54.361 kB`<br />`15.594 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/Switch.fixture.js'>Switch</abbr>                                                                                    |   `27.902 kB`<br />`8.647 kB` |
| <samp>react-switch</samp> <br /> <abbr title='bundle-size/SwitchField.fixture.js'>SwitchField</abbr>                                                                          |  `34.687 kB`<br />`10.541 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                                 | `129.423 kB`<br />`35.243 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                                            |  `42.971 kB`<br />`11.781 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                                 | `127.672 kB`<br />`32.893 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                                         |  `81.106 kB`<br />`20.161 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                                   |  `80.535 kB`<br />`20.005 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/Textarea.fixture.js'>Textarea</abbr>                                                                              |   `26.331 kB`<br />`8.581 kB` |
| <samp>react-textarea</samp> <br /> <abbr title='bundle-size/TextareaField.fixture.js'>TextareaField</abbr>                                                                    |  `36.469 kB`<br />`11.483 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e088f5010e9821b753a813ddf74aa2e89dae078e'>e088f5010e9821b753a813ddf74aa2e89dae078e</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/Avatar-mount_base.html">1575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/Avatar-mount.html">1580</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/Button-mount_base.html">1120</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/Button-mount.html">1138</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProvider-mount_base.html">1964</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProvider-mount.html">1950</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProviderWithTheme-mount_base.html">774</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProviderWithTheme-mount.html">772</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">727</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">702</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">750</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">773</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/InfoButton-mount_base.html">692</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/InfoButton-mount.html">682</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/MakeStyles-mount_base.html">2277</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/MakeStyles-mount.html">2278</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/Persona-mount_base.html">3628</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/Persona-mount.html">3580</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/SpinButton-mount_base.html">3015</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/26549/perf-test-react-components/SpinButton-mount.html">3041</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-field/docs/Spec.md [hunk 1]
```diff
@@ -2,176 +2,162 @@
...
 
-- **Accessibility**: By combining a base component with the field props into a single component, all of the accessibility props like `htmlFor` and `aria-describedby` are set correctly for "free".
-- **Simplicity**: All props related to the component (such as `label`, `id`, `validationState="error"`, etc.) are on the same component, rather than split between multiple components (like separate `Field` and `Input` components).
-- **Consistency**: All of the Field components share a common set of props for the label, validationState, hint, etc.
-- **Bundle size**: When the label and other field functionality is not needed, it is still possible to use the base components without pulling in unnecessary dependencies (like `Label` and the field styling).
+- **Accessibility**: All of the accessibility props like `aria-labelledby` and `aria-describedby` are set correctly on the child for "free".
+- **Simplicity**: Passing props down to the children allows any form control to be used as the child of Field.
+- **Consistency**: The Field component provides props the label, validationState, hint, etc. for any form control.
```
Comment (khmakoto) [inline packages/react-components/react-field/docs/Spec.md:29]: _nit:_

```suggestion
- **Consistency**: The Field component provides props like label, validationState, hint, etc. for any form control.
```

**Review by khmakoto — APPROVED** [review]

File: change/@fluentui-react-field-0653500c-e64a-4aa5-b0c8-f63390152e41.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-field-0653500c-e64a-4aa5-b0c8-f63390152e41.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 1 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=26549&commitId=110c7ebbd5dbc9d03eab71aa6b7063854a01f72d&env=prod)

✅ There was 0 screenshots added, 0 screenshots removed, 1842 screenshots unchanged, 0 screenshots with different dimensions and 1 screenshots with visible difference.


  <details><summary><b>unknown</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Image Converged.Default.normal.chromium.png|18239| Changed|

</details>


<div id="vrtCommentfluentuiv9"/>

**Pull request merged**